### DISCUSSION TITLE
Copilot Monthly Digest 📰

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#77548] [Copilot News and Announcements]
This month, we made some big improvements to GitHub Copilot! 🎉 **Copilot Chat is now powered by GPT-4** and we updated the model used to detect off-topic chat queries. In VSCode, we are announcing the public beta of code referencing. We also introduced “agents” and the ability to generate commit messages with Copilot. In addition, we improved the context for explaining code and updated the Copilot menu UI. In JetBrains IDEs, we introduced partial acceptance of code suggestions.

## Copilot Chat is powered by GPT–4 🚀 

We upgraded the Copilot Chat experience, bringing more accurate and useful code suggestions with OpenAI‘s GPT–4 model.

## Off–topic model improvements for Copilot Chat 🔒 

As part of our safety features, we‘ve improved our off-topic model to detect chat queries which do not relate to programming. This should result in significantly fewer filtered responses.

## Code referencing in VSCode is now in Public Beta 🚢 

In August, we announced the [Private Beta of code referencing in VSCode](https://github.blog/2023-08-03-introducing-code-referencing-for-github-copilot/). 

- This feature searches across billions of files on public GitHub repositories for code that matches a Copilot suggestion. Since then, we’ve heard your feedback, and we’re shipping with a new and redesigned experience. 
- One of the top points of feedback was that the original flow resulted in too many notifications. To fix this, if there’s a match, users will find its information displayed in the Copilot console log, including where the match occurred, any applicable licenses, and a deep link to learn more. 

Learn more about [Copilot code referencing](https://docs.github.com/copilot/using-github-copilot/finding-public-code-that-matches-github-copilot-suggestions) and let us know your thoughts in [this discussion](https://github.co/code-ref-public-beta-feedback).

## Introducing “agents” in Copilot Chat in VSCode 💬 

We have introduced a new capability called “agents” to enhance your interaction with Copilot Chat. Agents are like specialized experts who can assist you with specific tasks 🤝  . You can mention them in the chat using the `@` symbol. Each agent also supports slash commands. The slash commands you may have used before should now be used with an agent. For example, /explain is now `@workspace` /explain.

Read more in the [VSCode release notes](https://code.visualstudio.com/updates/v1_84#_chat-agents).

## Improved explanation context in Copilot Chat in VSCode 🆕 

You can ask Copilot Chat to explain a code selection in your active editor either through the `@workspace /explain` command or through the “Explain with Copilot” action in the context menu. Copilot Chat has now integrated implementations of referenced symbols, such as functions and classes, which leads to explanations that are more precise and useful. This works best across files when you have an extension contributing language services installed for one of the following languages: TypeScript/JavaScript, Python, Java, C#, C++, Go, or Ruby.

## Commit message generation using Copilot in VSCode[](#commit-message-generation-using-copilot-in-vscode) 🤖 

Copilot can now generate commit messages based on the pending changes using the new “sparkle” action in the Source Control input box.

![](https://github.blog/wp-content/uploads/2023/11/286359528-169dd29e-7a2c-4b17-a798-c6c515921471.gif?w=600&resize=600%2C324)

## Updated Copilot menu in VSCode 😮 

Our Copilot menu in VSCode is now more visible and aligned with our design for JetBrains IDEs. It is now easier to understand the current status of Copilot, access the various settings or documentation.

The new menu is displayed when clicking on the Copilot icon in the lower right corner in the statusbar of VSCode.

## JetBrains partial acceptance for code suggestions ✈️ 

The Copilot extension for JetBrains IDEs has leveled up! You now have the flexibility to incorporate code suggestions piece by piece, whether that’s word-by-word or line-by-line. Feel free to customize these shortcuts as you prefer. Happy coding!

![](https://github.blog/wp-content/uploads/2023/11/279422373-b54846be-ed10-4a60-9324-22fbdb0ecaa5.gif?resize=1024%2C599)

**Comment by nicomulderlawvu1** [comment]
Copilot Chat is meant to be General Available December 2023. Does this mean 1 Dec? Or some time, as the feature online is still marked as beta...?

**Reply by Akash1134** [reply]
GitHub Copilot Chat is now generally available for both [Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) and [Visual Studio](https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.VSGitHubCopilot), and is included in all [GitHub Copilot plans](https://github.com/features/copilot#pricing).

Read more about the announcement in this detailed [blog post](https://github.blog/2023-12-29-github-copilot-chat-now-generally-available-for-organizations-and-individuals/).

**Reply by nicomulderlawvu1** [reply]
Yes I saw thanks. Already enabled for our teams.

**Comment by Hefaistos68** [comment]
Why is it that the VS version is so behind the VScode version? Currently its pretty much useless inline in the C# code editor windows. Much of the functionality seen in the VScode version does not exist or work in VS. 
Care to explain? Is VScode now more important than VS?
